src/main/java/org/cloudbees/literate/jenkins/LiterateBuilder.java

```diff
@@ -185,8 +185,11 @@ public static boolean perform(AbstractBuild<?, ?> build, Launcher launcher, Buil
                         args.set(0, args.get(0).substring(2));   // trim off "#!"
                         commandLine = args.toArray(new String[args.size()]);
                     } else {
-                        Shell.DescriptorImpl shellDescriptor =
-                                Jenkins.getInstance().getDescriptorByType(Shell.DescriptorImpl.class);
+                        Jenkins j = Jenkins.getInstance();
+                        if (j == null) {
+                            throw new IllegalStateException(); // TODO 1.590+ getActiveInstance
```

Comment:
please add a message - if a user ever sees this having ("Jenkins is starting up or shutting down") is highly valuable.